File: Ch1-Fundamentals/0007-Leap-Year/main.js
function isLeapYear( incomingYear ) {
    if ( typeof(incomingYear) !== 'number' ) {
        console.log("`incomingYear` must be type number!");
        return false;
    }
    if (( incomingYear % 4 === 0 && incomingYear % 100 !== 0 ) || ( incomingYear % 400 === 0)) {
        console.log(`${incomingYear} is a leap year!`);
        return true;
    }
    console.log("NOT A LEAP YEAR!");
    return false;
}

isLeapYear(2000);
isLeapYear(1900);
isLeapYear('1900');
isLeapYear(2024);
isLeapYear(2021);

File: Ch1-Fundamentals/0007-Leap-Year/main.py
def is_leap_year( incomingYear ):
    if not isinstance( incomingYear, int ):
        print("`incomingYear` must be type number!")
        return False
    
    if (( incomingYear % 4 == 0 ) or ( incomingYear % 400 == 0)):
        print(f"{incomingYear} is a leap year!")
        return True
    print("NOT A LEAP YEAR!")
    return False

is_leap_year(2000)
is_leap_year(1900)
is_leap_year('2024')
is_leap_year(2024)
is_leap_year(2021)